### DISCUSSION TITLE
How to perform a double press on mobile: performIoHidEvent ?

### DISCUSSION THREAD

**Discussion by becph01** [appium/appium#16955] [Q&A]
Trying to emulate a double press on the IOS power button.

 Is it possible to do or is this a limitation on the appium driver?

This is what I'm referencing - https://github.com/appium/appium-xcuitest-driver#mobile-performiohidevent

"The event duration in float seconds. XCTest uses 0.005 for a single press event duration" 
What is the event float to do a double press? When trying to do two performIoHidEvents in quick duration - it doesn't recognize it as a double press.

**Comment by KazuCocoa** [comment]
Perhaps it cannot via XCTest framework.
https://github.com/appium/appium-xcuitest-driver#mobile-doubletap calls vanilla XCTest's doubleTap. Or W3C action can emulate the action.

**Comment by jainilparikh** [comment]
@becph01 , 

Hi were you able call the mobile-performiohideevent ? Can you share some code snippet?

**Reply by becph01** [reply]
I can call mobile-performiohideevent, but I can't get it to recognize it as a double press I don't understand what you'd like me to share?

@KazuCocoa so what would I need to do to a double tap or would it be impossible through appium at this point in time?

**Reply by KazuCocoa** [reply]
I tried to call `[self performDeviceEvent:event error:error]` in very short duration directly but it seems like the API (by OS) did not handle it as a double click. So it seems like OS's limitation in https://unix.superglobalmegacorp.com/xnu/newsrc/iokit/IOKit/hidsystem/IOHIDUsageTables.h.html for now.

**Comment by pralaykdas** [comment]
Hi, do we have any module similar to "performIoHidEvent" for Android platform?

**Reply by KazuCocoa** [reply]
Perhaps it is keyevent commands

**Comment by becph01** [comment]
Got it, that makes sense. Thank you for the clarification.

**Comment by sriteja777** [comment]
Hi did anyone able to crack this? I am also trying to emulate double side button press
